_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_vaapi.cpp

```diff
@@ -1078,6 +1080,15 @@ mfxStatus VAAPIEncoder::Execute(
             // 4. Segmentation map
 
             // segmentation map buffer is already allocated and filled. Need just to attach it
+            mfxU32 segCodedBuffer = task.m_pSegmentMap->idInPool;
+            if( segCodedBuffer < m_segMapQueue.size())
```

Comment:
I suggest to simplify this if:
```
MFX_CHECK( segCodedBuffer < m_segMapQueue.size(), MFX_ERR_UNDEFINED_BEHAVIOR);
m_segMapBufferId = m_segMapQueue[segCodedBuffer].surface;
```

Reply:
Done. Opened new pull request([#1153](https://github.com/Intel-Media-SDK/MediaSDK/pull/1153)) due to conflicts with master in this.

Reply:
@eshiryae actually you can rebase your branch locally and force-push it to your GitHub fork, it will update current PR and remove conflicts. Also it would be more convenient for reviewers